### DISCUSSION TITLE
if you get "failed (AREQ - ZDO - bindRsp after 10000ms)"  when pairing SNZB-02

### DISCUSSION THREAD

**Discussion by djp3k05** [Koenkk/zigbee2mqtt#9216] [General]
Hi,

After few days of trying to fix the pairing issues, I finally found the "solution". I will share it, hope to help someone.

All started when noticed that some of the SNZB-02 and SNZB-03 were not working anymore. I thought  that the battery had died. I had replaced the battery for one SNZB-02 and no luck. Then I force deleted it and tried to pair it again. Now I got that error than anyone got. 

Using the latest version (1.21.2-1) I got same errors as above. Restored 1.20.0-1 version from backup  and now I could pair back the device.

Now, I have to check the rest of not working devices.... if battery is dead or I have to pair them back again.

**Comment by RenoSkopek** [comment]
@djp3k05 that's nice. For me it works for 1 device of 3. However, I had to do this **50 times** at least.

Pls. could you help me how to downgrade to older version of Zigbee2mqtt? I'm new in Zigbee and also don't know Linux.

Command: sudo git clone https://github.com/Koenkk/zigbee2mqtt/archive/refs/tags/1.20.0.zip /opt/zigbee2mqtt
not work ...
How to do it?

Thank you

**Comment by dmasar** [comment]
can confirm problem in 1.22.2. when I rolled back to 1.20.0 snzb-02 gets paired.

**Comment by digdilem** [comment]
Am also getting this issue on a SNZB-02 on the current z2mqtt release (freshly pulled)

It used to work fine, but fell off my network during an update. (I didn't notice when). 

On re-pairing now, I get exactly as others are describing.

What _did_ work for me was to hold the button until the lights flash, then restart the Zigbee2mqtt container as soon as the "Sonoff removed" (Forget exact wording). 

When z2mqtt then starts, it discovered and configured the sensor automatically and it's now working as before.

**Reply by lumpynose** [reply]
@[digdilem](https://github.com/digdilem) That's very voodoo, but it also worked for me.  I'm running "Zigbee2MQTT version 1.23.0 (commit #6ea61bae)".  I ran zigbee2mqtt manually with npm using your voodoo sequence, then started it and saw the sonoff temperature sensor on the dashboard web page, then stopped zigbee2mqtt and start it how I usually do with systemctl (on linux) and it's still working.

Thanks for your voodoo.

**Reply by nvx** [reply]
wtf, this voodoo just worked for me too on 1.24.0 after failing a bunch of other times previously , @Koenkk thoughts on why this works?

**Reply by anthortic** [reply]
I can confirm this just worked for me also. So happy the answer was easier to find than I had anticipated. A weird workaround..... but a workaround it is. Current version: 1.24.0-1

**Reply by MichaelBitard** [reply]
Should we create an issue about that?

Could be related to https://github.com/Koenkk/zigbee2mqtt/issues/11911

**Reply by TimDowker** [reply]
Yes, I believe opening an issue would be good.

**Comment by IASolomin** [comment]
I have the same issue with 1 of 4 new sensors. 1.22.2 and 1.23 versions affected. 1.20.0 works fine.

**Reply by aldur11** [reply]
I have around 15 SNZB-03 motion sensors in different zigbee networks, using 1.22.2 and 1.23, and I've not had any problems. I've had them running for about 2 weeks. I'll report back here if they have any problems.

**Reply by lumpynose** [reply]
@[aldur11](https://github.com/aldur11) SNZB-03 is the motion sensor.  SNZB-02 is the temperature sensor.  Odd using SNZB for both of them.

**Reply by aldur11** [reply]
@lumpynose Yep, the OP mentioned both so I thought I'd chime in with my SNZB-03 experience.

**Reply by lumpynose** [reply]
Thanks.  I missed that -03 mention, obviously!

**Reply by lumpynose** [reply]
Which USB zigbee dongle are you using?  I'm wonder if it's specific to a dongle.  Mine is the Sonoff. Zigbee 3.0 Dongle Plus.

**Reply by digdilem** [reply]
Using slae.sh here. It was happy before, and since the voodoo above, happy since. 

No updates to the firmware of the affected temp sensor at any point. No updates to the controller firmware.

Suspect it's a little bug in how z2mqtt handles the automatic re-registering of these Sonoff devices that was introduced through another code change.

**Reply by IASolomin** [reply]
I have Sonoff Zigbee 3.0 Dongle Plus too

**Reply by aldur11** [reply]
I'm using the sonoff zigbee 3.0 dongle plus as coordinators for two of my zigbee networks and a TI dev board as a coordinator in my 3rd zigbee network. I'm also using some TI dev boards as routers and Tubes devices as routers. Getting close to to 300 zigbee devices total.

**Comment by suvl** [comment]
I do not have a backup to go back to, this is a new install. Any other way to fix this?

**Reply by nvx** [reply]
Have you tried this weird hack? https://github.com/Koenkk/zigbee2mqtt/discussions/9216#discussioncomment-2261788 it worked for me on the latest version.

**Reply by suvl** [reply]
better than that, clicking on the "reconfigure" button did the trick for me.
![image](https://user-images.githubusercontent.com/6018083/158651205-6c874bc6-adf7-4cff-a09e-247b37073569.png)

**Reply by TimDowker** [reply]
@suvl I tried the "reconfirgure" but got the same error.  I'm running version 1.24.0-1.

**Reply by redpixy** [reply]
@TimDowker same here, the "reconfirgure" shows AREQ - ZDO - bindRsp after 10000ms but it ends with device configured. I'm on [1.25.0](https://github.com/Koenkk/zigbee2mqtt/releases/tag/1.25.0) commit: [6f1460e](https://github.com/Koenkk/zigbee2mqtt/commit/6f1460e)

**Comment by lumpynose** [comment]
I just got the Conbee USB Zigbee dongle and the  SNZB-02 didn't have any problem joining with it on the first try whereas with the Sonoff USB Zigbee dongle the above voodoo is needed to get it to join.

**Comment by OzGav** [comment]
Just wanted to add the phrase Cygnett Smart Humidity and Temperature Sensor for future searchers. The voodoo hack worked for me as well. Thanks @digdilem

**Comment by mananoreboton** [comment]
I just tried the voodoo hack for SNZB-04 and SNZB-03 and worked like a charm XD. I have a Sonoff Zigbee 3.0 plus P btw.

**Reply by sherbibv** [reply]
Same, can someone tell us why does that work?

**Comment by creasysee** [comment]
FYI - I've applied the voodoo hack for SNZB-02D and it works fine. Z2M ver 1.31.0
What did I do:
1. I removed the device from a web UI -> main page -> red button "Remove device".
2. I restarted the Z2M service (systemctl restart zigbee2mqtt)
3. I waited for Z2M to load, refreshed the main page and pressed button "Permit join (All)"
4. I went to the location of the device (in another room) and pressed the button for 5 seconds.
Done.

Note: I have 2 devices SNZB-02D. The second device has no issue. Firmware versions identical (1.0.9 Build date 20230105). Battery 100% both.

Exactly the error looks like:
```
error 2023-12-13 00:20:05: Failed to configure 'Temperature_1_floor', attempt 1 (Error: Bind 0x3c2*********8f9/1 msTemperatureMeasurement from '0x0012************93/1' failed (AREQ - ZDO - bindRsp after 10000ms)
    at Timeout._onTimeout (/opt/zigbee2mqtt/node_modules/zigbee-herdsman/src/utils/waitress.ts:64:35)
    at listOnTimeout (internal/timers.js:557:17)
    at processTimers (internal/timers.js:500:7))
error 2023-12-13 00:20:34: Failed to configure 'Temperature_1_floor', attempt 2 (Error: Bind 0x3c2*********8f9/1 msTemperatureMeasurement from '0x0012************93/1' failed (AREQ - ZDO - bindRsp after 10000ms)
    at Timeout._onTimeout (/opt/zigbee2mqtt/node_modules/zigbee-herdsman/src/utils/waitress.ts:64:35)
    at listOnTimeout (internal/timers.js:557:17)
    at processTimers (internal/timers.js:500:7))
error 2023-12-13 06:53:00: Failed to configure 'Temperature_1_floor', attempt 3 (Error: Bind 0x3c2*********8f9/1 msTemperatureMeasurement from '0x0012************93/1' failed (AREQ - ZDO - bindRsp after 10000ms)
    at Timeout._onTimeout (/opt/zigbee2mqtt/node_modules/zigbee-herdsman/src/utils/waitress.ts:64:35)
    at listOnTimeout (internal/timers.js:557:17)
    at processTimers (internal/timers.js:500:7))
```
Thanks @digdilem

**Reply by WenderGalan** [reply]
**I'm having the same problem on my zigbee network, I don't know what else to do, it didn't work for me, does anyone else have the same problem?**

**Reply by cropse** [reply]
Same problem here, all firmare is up to date.

**Comment by tomeq82** [comment]
Latest possible z2m and same story. Even with the "voodoo" trick it worked for a while eg. read one bundle of all measurements and then became dead silent. I tried everything, removing forcibly, giving different names, leaving default, removing battery, changing battery to different two ones, pushing button long, short, rapidly - whatever, restarting moving closer to coordinator etc. etc. Nothing seem to wake up this sensor. Suprisingly - this one is being least problematic (!) one in HA direct integration (ZHA) while others are missing readings (eg. Thuya TS0201) this one seem to be stable as rock. But not in Z2M. What the actual heck?

**Comment by PiotrDuz** [comment]
I confirm that @digdilem 's answer helps. My sensors (  SNZB-02P ) were added, but were reporting temperatures slowly and with large threshold.
When wanted to change the reporting min change I was getting the error:  AREQ - ZDO - bindRsp after 10000ms 

Holding the button until z2m tells that sensor dropped and then restarting the z2m integration solved the issue :) I can now set new values for reporting periods etc

**Comment by m-aar-ten** [comment]
Can confirm that removing and repairing did work to change the reporting values, but now that I want to change them again a day later I get the same error again..

**Comment by adigundecha** [comment]
Any such voodoo for sonoff dongle P? I am stuck by this error and all my parasoll are behaving weirdly.

**Reply by andrei-lazarov** [reply]
Parasoll are known to be problematic. I think the fix is to configure more frequent reporting
- Please see latest replies in https://github.com/Koenkk/zigbee2mqtt/issues/22579
